app/services/registration_csv_converter.rb

```diff
@@ -86,4 +90,27 @@ def dro_type(content_type)
       Cocina::Models::ObjectType.object
     end
   end
+
+  def structural(row)
+    {}.tap do |structural|
+      structural[:isMemberOf] = [row['collection']] if row['collection']
+      structural[:hasMemberOrders] = [{ viewingDirection: row['reading_order'] }] if row['reading_order'].present?
+    end
+  end
+
+  def access(row)
+    {}.tap do |access|
+      access[:view] = row.fetch('rights_view')
+      access[:download] = row.fetch('rights_download')
+      access[:location] = row.fetch('rights_location') if [access[:view], access[:download]].include?('location-based')
+      access[:controlledDigitalLending] = controlled_digital_lending?(access)
```

Comment:
This doesn't seem right.  It's not automatically CDL if stanford/none.  But it should optionally allow CDL to be set if these preconditions are met.

Reply:
@jcoyne Suggestions on a field or how to handle that? I was basing this on https://github.com/sul-dlss/argo/blob/223ef77fa4f559387d4973d568132f5523968245/app/services/cocina_dro_access.rb which uses `rights == 'cdl-stanford-nd'` do set controlledDigitalLending to true, but I don't see anything going the other way.

Reply:
Perhaps @andrewjbtw can weigh in on how he may prefer to pass in CDL via the spreadsheet.

Reply:
Could we cover this with an optional column in the CSV for `rights_controlledDigitalLending`? But I would like to avoid making people enter "false" when they don't want CDL. It's such an exceptional status that I'd like to set things up so people only indicate when it's "true".

That could mean:

- only require the column for CDL if one of the items on the spreadsheet will have "true"
- if the column is present and the cell for a particular row is blank, treat that as "false"
- if the column is present and the cell for a particular row has "true", set CDL
- if the column is present and all cells are blank, treat that as "false"

Reply:
And to be explicit, the "view:stanford" and "download:none" must also be set to be valid for CDL.

Reply:
I think an optional column with `view:stanford` and `download:none` validation would be fine. :+1: